MantidQt/MantidWidgets/src/InstrumentView/InstrumentWidgetPickTab.cpp

```diff
@@ -967,6 +989,61 @@ void ComponentInfoController::displayComparePeaksInfo(
   m_selectionInfoDisplay->setText(QString::fromStdString(text.str()));
 }
 
+void ComponentInfoController::displayAlignPeaksInfo(
+    const std::vector<Mantid::Kernel::V3D> &planePeaks,
+    const Mantid::Geometry::IPeak *peak) {
+
+  if (planePeaks.size() < 3)
+    return;
+
+  const auto pos1 = planePeaks[0];
+  const auto pos2 = planePeaks[1];
+  const auto pos3 = planePeaks[2];
+
+  // find vectors in plane & plane normal
+  auto u = pos2 - pos1;
+  auto v = pos3 - pos1;
+  auto n = u.cross_prod(v);
+
+  // find projection of beam direction onto plane
+  // this is so we always orientate to a common reference direction
+  const auto instrument = peak->getInstrument();
+  const auto samplePos = instrument->getSample()->getPos();
+  const auto sourcePos = instrument->getSource()->getPos();
+  const auto beam = samplePos - sourcePos;
+  const auto proj = beam - n * (beam.scalar_prod(n) / (pow(n.norm(), 2)));
```

Comment:
1) Sure. 2) Good spot. It is but it's unlikely. I can put something in to handle that case.